vlog4j-parser/src/test/java/org/semanticweb/vlog4j/syntax/parser/RuleParserTest.java

```diff
@@ -438,4 +454,82 @@ public void testCustomDatatype() throws ParsingException {
 		DatatypeConstant result = (DatatypeConstant) literal.getConstants().toArray()[0];
 		assertEquals(constant, result);
 	}
+
+	@Test
+	public void testGetFacts() throws ParsingException, IOException {
```

Comment:
instead of hardcoding the expected chase result, for an integration test I suggest the following scenario:
1. you add some data+rules in a KB `kb1`
2. you reason on `kb1`
3. you query `kb1` for every single predicate you have in the kb, and save all of these queries&results in a collection `res` (for example, a HashMap)
4. you export the chase result of  `kb1` to a file, and then parse it into  `kb2` 
5. you reason on `kb2` (note that there are no rules in `kb2`)
6. you query `kb2`  for all the queries in `res`, and check that you get the same results as when you queried `kb1`.
Blanks introduced to satisfy existential rules may require some pre-processing, I am not sure if they will be round-tripped into AbstractConstants.

Reply:
maybe we can discuss this in the meeting because I get the feeling that this is the same as implementing the method all over again, and I don't see why what we have now is not ok